src/pkg/egress/syslog/triggerTimer.go

```diff
@@ -0,0 +1,39 @@
+package syslog
+
+import "time"
+
+type TriggerTimer struct {
+	trigger chan int
+	running bool
+}
+
+type Timer interface {
+	Start(d time.Duration, f func())
+}
+
+func NewTriggerTimer() Timer {
+	return &TriggerTimer{
+		running: false,
```

Comment:
`trigger: make(chan int)` or risk a nil chan runtime error.

Reply:
The issue here is, that the timer implementation has no good way of interrupting by another source.
I need both:
- interrupt by time elapsed
- interrupt by buffer size

The old logic was not catching all possible states that could occur correctly, and was not really able to be reconciled in a way to fix the state issues without looking totally unreadable.
Therefore, I opted for factoring it out into a struct handling all the state and trigger changes.
If this is unsafe, please advise on how to create this one more safe, as I am not the most knowledgeable about go concurrency at the moment.

Reply:
I was just commenting that the zero type of a `chan int` is `nil`, and so you can't perform channel operations on it until you instantiate it with `make`.

Let's switch to [this comment](https://github.com/cloudfoundry/loggregator-agent-release/pull/491#discussion_r1592717135).